apisix/init.lua

```diff
@@ -629,6 +630,14 @@ end
 function _M.http_header_filter_phase()
     core.response.set_header("Server", ver_header)
 
+    local status_code = tonumber(get_var("upstream_status"))
+    if status_code then
+        if status_code >= 500 and status_code <= 599 then
```

Comment:
could we always respond upstream status unless it's greater than 200 ？

Reply:
The main purpose here is to distinguish whether the 5xx status code comes from APISIX or upstream. If we want to record other response status codes, we need more discussion.